### ISSUE TITLE
Proper ServerRPC to make it impossible to use ServerCmd's from Clients by the Ingame console

### ISSUE THREAD

**Issue by Adesii** [Facepunch/sbox-issues#648]
**What can't you do?**
Proper ServerRPC's without them showing up in the ingame Console. And only callable from functions.

**How would you like it to work?**
ServerCmd's show up in the Console so they can be used for stuff like this:
![10,08_03,11 09_466](https://user-images.githubusercontent.com/37816311/128793572-4e7c4138-6a39-4e44-9697-d35aaaf3028a.png)
![10,08_03,11 32_467](https://user-images.githubusercontent.com/37816311/128793574-1d7a0a8f-bd7c-47d7-ad5a-a37f8b50b166.png)
![10,08_03,34 33_46d](https://user-images.githubusercontent.com/37816311/128794942-e9072f04-6d6d-4c6a-9735-85b745e1a08c.png)
Which is great for Testing but not so great for Actual gameplay.

There should be a different method like it works for ClientRPC and ClientCmd.
this would make it way easier to prevent cheating by calling commands.

**What have you tried?**

currently the only way to avoid this from happening is by using ridiculous ServerCmd names that the ingame console can't use.
basically with special characters.

like this (over the top example : `\"` this alone works too):
![10,08_03,15 41_46a](https://user-images.githubusercontent.com/37816311/128793614-7bb53e85-a65b-4158-9823-2196d4be6fa2.png)
![10,08_03,15 31_468](https://user-images.githubusercontent.com/37816311/128793616-136b0a37-546b-4670-aef0-20b63541a74a.png)

but this makes it impossible to call the method from Clients.

**Comment by Rohansi** [comment]
If your goal is to prevent cheating then this will not work. You want the server to be calling `GiveResource` without relying on the client. And for easy testing you can `[ServerCmd]` expose it but with a check to see if the player is allowed to cheat before actually giving them anything.

**Comment by Adesii** [comment]
It is less about preventing cheating and instead of just making it harder.
if they can't be called from the console itself that would already prevent a good chunk of people from trying to cheat.
but currently its just laid out open.

i could of course just make it everything server sided. like the whole process of finding out how much to give etc.
but in a peer to peer game that would sooner or later create a bottleneck. and just make the whole experience worse for everyone involved.

i technically just want a way to not let these commands be called from the console. 
if that is a ServerRPC or just a property on ServerCmd to hide it. idc about that.

**Comment by thegrb93** [comment]
I need ServerRpc to send binary data to server. ServerCmd botches it.

**Comment by ogniK5377** [comment]
If it's a need of the ServerRPC attribute, this is a duplicate of #191

**Comment by robotboy655** [comment]
Duplicate of https://github.com/sboxgame/issues/issues/191
